tests/BUILD.bazel

```diff
@@ -54,6 +54,30 @@ load("@io_bazel_rules_go//go:def.bzl", "go_binary")
     ),
 ]
 
+# Test nixopts location expansion
+sh_test(
+    name = "location-expansion-test",
+    srcs = ["location_expansion.sh"],
+    args = [
+        "$(POSIX_DIFF)",
+        "$(rootpath //:nixpkgs.json)",
+        "$(rootpath //:nixpkgs.nix)",
+        "$(rootpath //tests:relative_imports/nixpkgs.nix)",
+        "$(rootpath @nixpkgs_location_expansion_test//:out/nixpkgs.json)",
+        "$(rootpath @nixpkgs_location_expansion_test//:out/nixpkgs.nix)",
+        "$(rootpath @nixpkgs_location_expansion_test//:out/relative_imports.nix)",
+    ],
+    data = [
+        "//:nixpkgs.json",
+        "//:nixpkgs.nix",
+        "//tests:relative_imports/nixpkgs.nix",
+        "@nixpkgs_location_expansion_test//:out/nixpkgs.json",
+        "@nixpkgs_location_expansion_test//:out/nixpkgs.nix",
+        "@nixpkgs_location_expansion_test//:out/relative_imports.nix",
+    ],
+    toolchains = ["@rules_sh//sh/posix:make_variables"],
+)
```

Comment:
This test is extremely convoluted, it has so many moving parts that it’s hard to understand what it does or why. Needs either a more detailed description of what is done and why, or be split up into more tests that each do a lot less.

Reply:
For example, the location expansion stuff is just a string replacement, and the algo could be tested as a unit-test entirely; which would mean the integration test needs to test less.

Reply:
With the factored out pure parsing and label resolution functions it is now possible to define unit tests for these, which I've done. I've simplified this integration test to only test that the mapping to paths works correctly for a file in the main repository and for one in an external repository.